File: 编程/leetcode/1470-[E]-重新排列组合/solution.py
from typing import List


class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        new_nums = []
        for i in range(n):
            new_nums.extend([nums[i], nums[i+n]])
        return new_nums


if __name__ == '__main__':
    test = Solution()
    print(test.shuffle([2,5,1,3,4,7], 3))
    print(test.shuffle([1,2,3,4,4,3,2,1], 4))

File: 编程/leetcode/1470-[E]-重新排列组合/solution.rs
struct Solution;

impl Solution {
    pub fn shuffle(nums: Vec<i32>, n: i32) -> Vec<i32> {
        let mut new_nums: Vec<i32> = Vec::new();
        for i in 0..n {
            new_nums.push(nums[i as usize]);
            new_nums.push(nums[(i+n) as usize]);
        };
        new_nums
    }
}

fn main() {
    println!("{:?}", Solution::shuffle(vec![2,5,1,3,4,7], 3))
}